smack-experimental/src/test/java/org/jivesoftware/smackx/reactions/ReactionTest.java

```diff
@@ -0,0 +1,164 @@
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.mock;
+
+/**
+ * Tests related to managing reactions in Smack XMPP.
+ * These tests cover the creation and manipulation of reaction elements,
+ * including adding reactions to messages, XML parsing, and verifying
+ * invalid restrictions.
+ *
+ * @author Ismael Nunes Campos
+ * @since 2025
+ */
+public class ReactionTest extends SmackTestSuite {
+
+    /**
+     * Tests parsing of a reactions XML and validates the extracted data.
+     *
+     * This test validates the creation of a `ReactionsElement` from an XML
+     * and checks if the message ID and emoji reactions are correctly extracted.
+     *
+     * @throws XmlPullParserException If an error occurs during XML parsing.
+     * @throws IOException If an I/O error occurs during parsing.
+     * @throws SmackParsingException If a failure occurs while parsing the XML.
+     */
+    @Test
+    void testReaction() throws XmlPullParserException, IOException, SmackParsingException {
+        String xml = "<reactions xmlns='urn:xmpp:reactions:0' id='msg-id-123'>" +
+                "<reaction>👍</reaction>" +
+                "<reaction>❤️</reaction>" +
+                "</reactions>";
+
+        XmlPullParser parser = TestUtils.getParser(xml);
+
+        ReactionsElementProvider provider = new ReactionsElementProvider();
+        ReactionsElement reactionsElement = provider.parse(parser);
+
+        assertEquals("msg-id-123", reactionsElement.getId());
+        assertEquals(2, reactionsElement.getReactions().size());
+        assertEquals("👍", reactionsElement.getReactions().get(0).getEmoji());
+        assertEquals("❤️", reactionsElement.getReactions().get(1).getEmoji());
+    }
+
+    /**
+     * Tests adding reactions to a message.
+     *
+     * This test ensures that the `addReactionsToMessage` method correctly
+     * adds reactions to a message and that the reactions are correctly
+     * reflected in the message's extension elements.
+     *
+     * @throws Exception If an error occurs during message handling or assertions.
+     */
+    @Test
+    public void testAddReactionsToMessage() throws Exception {
+
+        List<String> emojis = Arrays.asList("❤️", "❤️");
+        String messageId = "1234";
+
+        MessageBuilder messageBuilder = StanzaBuilder.buildMessage();
+        Message message = messageBuilder
+                        .setBody("Hello")
+                        .ofType(Message.Type.chat)
+                        .to("teste@domain.com")
+                        .build();
+
+        ReactionsManager.addReactionsToMessage(message, emojis, messageId, null);
+
+        ReactionsElement reactionsElement = (ReactionsElement) message.getExtensionElement(ReactionsElement.ELEMENT, ReactionsElement.NAMESPACE);
```

Comment:
Casts are bad, better use the type safe approach (may need adjustments)
```suggestion
        ReactionsElement reactionsElement = message.getExtensionElement(ReactionsElement.class);
```